lms/static/js/learner_dashboard/views/entitlement_unenrollment_view.js

```diff
@@ -0,0 +1,128 @@
+                            window.accessible_modal(
+                                '#' + $trigger.attr('id'),
+                                view.closeButtonSelector,
+                                '#' + view.$el.attr('id'),
+                                view.mainPageSelector
+                            );
+                        }
+                    });
+                },
+
+                handleTrigger: function(event) {
+                    var $trigger = $(event.target),
+                        courseName = $trigger.data('courseName'),
+                        courseNumber = $trigger.data('courseNumber'),
+                        apiEndpoint = $trigger.data('entitlementApiEndpoint');
+
+                    this.resetModal();
+                    this.setHeaderText(courseName, courseNumber);
+                    this.setSubmitData(apiEndpoint);
+
+                    if (window.edx && window.edx.dashboard && window.edx.dashboard.dropdown) {
+                        window.edx.dashboard.dropdown.toggleCourseActionsDropdownMenu(event);
+                        this.$el.css('position', 'fixed');
+                    }
+                },
+
+                handleSubmit: function() {
+                    var apiEndpoint = this.$submitButton.data('entitlementApiEndpoint');
+
+                    if (apiEndpoint === undefined) {
+                        this.setError(gettext('Error: cannot process unenrollment request.'));
+                        return;
+                    }
+
+                    $.ajax({
+                        url: apiEndpoint,
+                        method: 'DELETE',
+                        complete: this.onComplete.bind(this)
+                    });
+                },
+
+                resetModal: function() {
+                    this.$submitButton.removeData();
+                    this.$submitButton.prop('disabled', false);
+                    this.$headerText.empty();
+                    this.$errorText.removeClass('entitlement-unenrollment-modal-error-text-visible');
+                    this.$errorText.empty();
+                },
+
+                setError: function(message) {
+                    this.$submitButton.prop('disabled', true);
+                    this.$errorText.empty();
+                    HtmlUtils.setHtml(
+                        this.$errorText,
+                        message
+                    );
+                    this.$errorText.addClass('entitlement-unenrollment-modal-error-text-visible');
+                },
+
+                setHeaderText: function(courseName, courseNumber) {
+                    this.$headerText.empty();
+                    HtmlUtils.setHtml(
+                        this.$headerText,
+                        HtmlUtils.interpolateHtml(
+                            gettext('Are you sure you want to unenroll from {courseName} ({courseNumber})? You will be refunded the amount you paid.'),
```

Comment:
@catong could you please confirm that this message is acceptable? This will be displayed in the entitlement unenrollment modal above the 'unenroll' button. The course run version of the modal has more variation with regard to this text, for things like the certificate type and course mode, but for entitlements I was hoping to keep it simple.

Reply:
@amangano-edx Text looks good to me. 
But can you please clarify when the gear icon appears? I think: when a learner is not eligible for a refund, the "unenroll" option is removed from the dropdown menu, but the gear icon should still be available (with other existing options e.g. "Email Settings"). Is this your understanding too?